core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1994,11 +1994,15 @@ public void run() {
           }
         }
 
-        delayedShutdownTask = syncContext.schedule(
-            new LogExceptionRunnable(new ShutdownSubchannel()),
-            SUBCHANNEL_SHUTDOWN_DELAY_SECONDS, TimeUnit.SECONDS,
-            transportFactory.getScheduledExecutorService());
-        return;
+        ScheduledExecutorService scheduledExecutorService =
+            transportFactory.getScheduledExecutorService();
+        if (!scheduledExecutorService.isShutdown()) {
```

Comment:
What is this about? If it is shut down, then something is broken before this point, as this is a required resource for the life of the transport. And why is it a good idea to simply not shut down the subchannel if we can't schedule?